Sources/Datasets/datasets/image/MNIST.swift

```diff
@@ -0,0 +1,113 @@
+import Foundation
+import S5TF
+import TensorFlow
+
+public struct MNIST: S5TFDataset {
+    typealias DataLoader = MNISTDataLoader
+    public static var train: MNISTDataLoader {
+        get {
+            return MNISTDataLoader(split: .train)
+        }
+    }
+    public static var test: MNISTDataLoader {
+        get {
+            return MNISTDataLoader(split: .test)
+        }
+    }
+
+    public static var info = mnistInfo
+
+    private init() {}
+}
+
+fileprivate let mnistInfo = S5TFDatasetInfo(
+    name: "mnist",
+    version: "0.0.1",
+    description: "The MNIST database of handwritten digits. 60000 train examples and 10000 test examples with image and label features.",
+    homepage: URL(string: "http://yann.lecun.com/exdb/mnist/")!,
+    numTrain: 60000,
+    numTest: 10000,
+    numFeatures: 10
+)
+
+public struct MNISTDataLoader: S5TFDataLoader {
+    private var index = 0
+    public let count: Int
+
+    public let split: S5TFSplit
+    public let batchSize: Int?
+
+    private let data: Tensor<Float>
+    private let labels: Tensor<Int32>
+
+    private init(split: S5TFSplit, data: Tensor<Float>, labels: Tensor<Int32>, batchSize: Int? = nil) {
+        self.split = split
+        self.data = data
+        self.labels = labels
+        self.batchSize = batchSize
+        self.count = data.shape.dimensions.first!
+    }
```

Comment:
This is the private loader which makes `S5TFDataLoader` conform to the protocol. It will be an hard requirement in the future. This means that `split` cannot be used because it would conflict with the protocol definition.

Plus you do (and should) not use `split` anywhere in your code so please remove the property.